tests/src/core/base64Test.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * base64Test.cpp
+ *
+ * Copyright (c) 2022 Kevin Zander <veratil/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "QTestSuite.h"
+
+#include "base64.h"
+
+class Base64Test : QTestSuite
+{
+    Q_OBJECT
+private slots:
+    void Base64Tests()
+    {
+        using namespace lmms::base64;
+
+        // Test Vectors from RFC 4648 Section 10
+        std::vector<std::pair<std::string, std::string>> test_vectors{
+            {"", ""},
+            {"f", "Zg=="},
+            {"fo", "Zm8="},
+            {"foo", "Zm9v"},
+            {"foob", "Zm9vYg=="},
+            {"fooba", "Zm9vYmE="},
+            {"foobar", "Zm9vYmFy"},
+        };
+        for (auto vector : test_vectors)
```

Comment:
Since we use C++17, something like this is also possible:
```suggestion
        for (const auto& [original, encoded] : test_vectors)
```

Reply:
Alternatively, one could use the technique described in https://doc.qt.io/qt-5/qttestlib-tutorial2-example.html, which has the advantage that the collection of pairs is exposed to the test framework. I'm not sure how well it plays with non-Qt types though, and it is a little more verbose.

Reply:
I went the QTest route. Very cool how it works.